app/controllers/email_confirmations_controller.rb

```diff
@@ -48,6 +59,8 @@ def webauthn_update
     confirm_email
   rescue WebAuthn::Error => e
     login_failure(e.message)
+  ensure
+    delete_mfa_expiry_session
```

Comment:
this seems to be repeated a few times, I think it could be simplified to `after_action :delete_mfa_expiry_session, except: [...]`

Reply:
Good idea! Fixed in most recent commit.